nexus/src/db/datastore/sled.rs

```diff
@@ -62,4 +63,25 @@ impl DataStore {
             .await
             .map_err(|e| public_error_from_diesel_pool(e, ErrorHandler::Server))
     }
+
+    pub async fn random_sled(
+        &self,
+        opctx: &OpContext,
+    ) -> OptionalLookupResult<Sled> {
+        opctx.authorize(authz::Action::Read, &authz::FLEET).await?;
+        use db::schema::sled::dsl;
+
+        sql_function!(fn random() -> Text);
```

Comment:
"how is this working" is never a dumb question, considering it usually reveals a bug :) I threw in a `diesel::debug_query` for that, and it shows:

```sql
SELECT "sled"."id", "sled"."time_created", "sled"."time_modified", "sled"."time_deleted", "sled"."rcgen", "sled"."rack_id", "sled"."ip", "sled"."port", "sled"."last_used_address" FROM "sled" WHERE ("sled"."time_deleted" IS NULL) ORDER BY random() LIMIT $1 -- binds: [1]
```

It looks like it's working only because the `random()` is used in the ORDER BY, and not returned by the SELECT, so it never type checks. Changed to Float in 99f346e7